core/src/main/java/com/netease/arctic/utils/ArcticDataFiles.java

```diff
@@ -114,6 +114,10 @@ public static GenericRecord data(PartitionSpec spec, String partitionPath) {
               field.name().equals(parts[0]),
           "Invalid partition: %s", partitions[i]);
 
+      if ("null".equals(parts[1])) {
```

Comment:
If the value of the partition is the string "null", will it cause misjudgment here?

Reply:
The PartitionSpec#partitionToPath method of Iceberg cannot distinguish between these two situations. If the existing pattern is used, then it is not possible to avoid this problem.